models/repo/repo.go

```diff
@@ -726,19 +726,23 @@ func GetRepositoriesMapByIDs(ids []int64) (map[int64]*Repository, error) {
 	return repos, db.GetEngine(db.DefaultContext).In("id", ids).Find(&repos)
 }
 
-// IsRepositoryExist returns true if the repository with given name under user has already existed.
-func IsRepositoryExist(ctx context.Context, u *user_model.User, repoName string) (bool, error) {
-	has, err := db.GetEngine(ctx).Get(&Repository{
-		OwnerID:   u.ID,
-		LowerName: strings.ToLower(repoName),
-	})
+// IsRepositoryModelOrDirExist returns true if the repository with given name under user has already existed.
+func IsRepositoryModelOrDirExist(ctx context.Context, u *user_model.User, repoName string) (bool, error) {
```

Comment:
We should move the directory check to the service layer and leave the database check here.

Reply:
I do not see it is feasible. There is a `RepoPath` for directory in this package. Moving `RepoPath` would be big refactoring.

Reply:
> I do not see it could be done easily at the moment. There is a `RepoPath` for directory in this package. Moving `RepoPath` would be big refactoring.

Yes, I have a PR about git storage to try to move `RepoPath` just return a relative path.